### PR TITLE
Document code layout with READMEs

### PR DESCRIPTION
**Pull request by slipher**

### CODE REVIEW

**Comment by illwieckz** [comment]
Maybe name them `README.md` so we can use markdown and are more code forge friendly (GitHub, GitLab, etc. know how to display a `README.md` at the root of a folder)?

**Comment by slipher** [comment]
I don't want to use markdown, because I want it to be readable on the developer's own filesystem.

**Comment by illwieckz** [comment]
The purpose of Markdown is meant to be readable without Markdown viewer.

**Comment by slipher** [comment]
OK I added the extensions so it won't be displayed with typewriter font...

File: src/common/IPC/README.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+This directory contains code relating to Inter-Process Communication (i.e. between the engine and a
```
Comment (illwieckz) [inline src/common/IPC/README.md:1]: Why not removing the repetitive and obvious `This directory contains`? It looks like all directories are directories containing something, so maybe we can just start with what it contains, this way:

```
Code relating to Inter-Process Communication (i.e. between the engine and a gamelogic VM).
```

File: src/common/cm/README.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+This directory contains the library for loading, traversing, and querying BSPs. A .bsp file is the
+compiled form of a map. The renderer, the server, and the server and client gamelogics all use it.
```
Comment (illwieckz) [inline src/common/cm/README.md:2]: Are those two `server` mentions about different things, or is it a duplicate?

File: src/engine/README.md [hunk 3]
```diff
@@ -0,0 +1,2 @@
+This directory primarily contains code for the engine. That is, the main program that runs
```
Comment (illwieckz) [inline src/engine/README.md:1]: If we remove `This directory primarily`, we may also remove `That is,`:

```
Code for the engine: the main program that runs […]
```

File: src/common/cm/README.md (hunk 2 above)
Comment (slipher) [reply src/common/cm/README.md:2]: Yes server and server gamelogic are different things.

File: src/common/IPC/README.md (hunk 1 above)
Comment (slipher) [reply src/common/IPC/README.md:1]: I don't consider it so obvious as to go without saying. Plus, it changes the meaning. `Code for the engine` suggests the *entire* source of the engine, while `This directory contains code for the engine` makes it clearer that it doesn't necessarily mean all engine code.

Comment (illwieckz) [reply src/common/IPC/README.md:1]: OK

**Review by illwieckz — APPROVED** [review]
LGTM.

**Comment by bmorel** [comment]
LGTM. Adding doc, without any risk of introducing bugs, that's a great PR!

**Pull request merged**